src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+// отримуємо коллекцію усіх елементів списку
+const headline = document.querySelectorAll('li');
+
+function createSpan(el) {
+  const spanEl = document.createElement('span');
+  const textNode = el.firstChild;
+
+  el.firstChild.parentNode.insertBefore(spanEl, textNode);
+  spanEl.appendChild(textNode);
+}
+
+for (const el of headline) {
+  if (el.children.length > 0) {
+    createSpan(el);
+  }
+}
+
+const spanArray = [...document.querySelectorAll('span')];
+
+for (const li of spanArray) {
+  li.addEventListener('click', (e) => {
+    const children = li.nextElementSibling;
```

Comment:
The variable `children` is assigned `li.nextElementSibling`, which assumes that every `li` element has a next sibling. If this is not the case, `children` will be `null`, and trying to access `children.style` will result in a runtime error. Consider adding a check to ensure `children` is not `null` before accessing its `style` property.